src/hats/pixel_math/healpix_shim.py

```diff
@@ -1,41 +1,85 @@
 from __future__ import annotations
 
+import math
+
 import astropy.units as u
+import cdshealpix
 import healpy as hp
 import numpy as np
-from astropy.coordinates import SkyCoord
+from astropy.coordinates import Latitude, Longitude, SkyCoord
 
 # pylint: disable=missing-function-docstring
 
 ## Arithmetic conversions
 
 
-def npix2order(param):
-    return hp.nside2order(hp.npix2nside(param))
+MAX_HEALPIX_ORDER = 29
+
+
+def is_order_valid(order: int) -> bool:
+    return np.all(0 <= order) and np.all(order <= MAX_HEALPIX_ORDER)
+
+
+def npix2order(npix: int) -> int:
+    if npix <= 0:
+        raise ValueError("Invalid value for npix")
+    order = int(math.log2(npix / 12)) >> 1
+    if not is_order_valid(order) or not 12 * (1 << (2 * order)) == npix:
+        raise ValueError("Invalid value for npix")
+    return order
+
+
+def order2nside(order: int) -> int:
+    if not is_order_valid(order):
+        raise ValueError("Invalid value for order")
+    return 1 << order
+
+
+def order2npix(order: int) -> int:
+    if not is_order_valid(order):
+        raise ValueError("Invalid value for order")
+    return 12 * (1 << (2 * order))
+
 
+def nside2resol(nside: int, arcmin: bool = False) -> float:
```

Comment:
Can we also provide an `order2resol` method? We only use the `nside2resol` with a call to `order2nside`.

Same with `nside2pixarea`.

See description in https://github.com/astronomy-commons/hats/issues/423